### DISCUSSION TITLE
请问支持kv 和kkv这种表形式吗

### DISCUSSION THREAD

**Discussion by mykittylover** [milvus-io/milvus#33865] [Q&A and General discussion]
请问支持kv， kkv 这种表结构么

**Comment by yhmo** [comment]
**请下次提问时尽量使用英文。**

kv, kkv表结构具体指什么？能否举个例子

**Reply by mykittylover** [reply]
for example  { "sku1": "sku2:0.8, sku3:0.9, sku4:1.0"} for kv
{"sku1" : {{ "sku2" : "0.8"}, {"'sku3" : "0.9"}}}

**Reply by yhmo** [reply]
Example:
```
from pymilvus import (
    connections,
    FieldSchema, CollectionSchema, DataType,
    Collection,
    utility,
)


HOST = '127.0.0.1'
PORT = '19530'


if __name__ == '__main__':
    connections.connect(host=HOST, port=PORT)
    print(utility.get_server_version())

    collection_name = "test_collection"
    if utility.has_collection(collection_name):
        utility.drop_collection(collection_name)

    fields = [
        FieldSchema(name="id", dtype=DataType.INT64, is_primary=True),
        FieldSchema(name="vector", dtype=DataType.FLOAT_VECTOR, dim=4),
        FieldSchema(name="sku1", dtype=DataType.JSON),
    ]
    collection = Collection(name=collection_name, schema=CollectionSchema(fields))

    collection.insert({"id": 1, "vector": [1, 1, 1, 1], "sku1": {"sku2": "0.8", "'sku3": "0.9"}})
    collection.insert({"id": 2, "vector": [2, 2, 2, 2], "sku1": {"sku2": "1.2", "'sku3": "1.3"}})
```

**Reply by mykittylover** [reply]
It's not quite accurate. What I wanted to express was regarding the support for KV (Key-Value) tables and KKV (Key-Key-Value) tables. The example you provided does not relate to a KV or KKV table. In a KV table, 'K' stands for the primary key, and 'V' represents the attributes.

**Comment by xiaofan-luan** [comment]
目前用json实现的，效率不是很高，接下来有计划支持原生的map数据类型。

**Comment by xiaofan-luan** [comment]
> 请问支持kv， kkv 这种表结构么

使用场景是推荐么？

**Reply by mykittylover** [reply]
嗯，是的推荐的场景

**Reply by mykittylover** [reply]
是不是目前没有计划支持这种kv表的形式呀

**Reply by xiaofan-luan** [reply]
你说的这个kkv需要更新么？还是只需要支持查询的时候需要 where pk = 1 && a["key"] == "test"

**Reply by xiaofan-luan** [reply]
后者可以通过map支持，还是比较简单的

**Reply by mykittylover** [reply]
> 你说的这个kkv需要更新么？还是只需要支持查询的时候需要 where pk = 1 && a["key"] == "test"

需要更新的，天级别更新

**Comment by xiaofan-luan** [comment]
这个支持可以通过map做。更新会比较复杂，主要是性能考虑

1.  可以通过批量导入支持 
2. 不过会涉及到老数据的删除，如果大量更新（每天更新老数据会造成大量删除）
所以在你们的场景下，大概会有多少删除场景？
